service_management/app/models/analysis/ResidualMaterials.scala

```diff
@@ -0,0 +1,52 @@
+package models.analysis
+
+import play.api.libs.json._
+
+object ResidualMaterials {
+
+  sealed trait ResidualMaterial {
+    val identity: Int
+  }
+
+  object ResidualMaterial {
+    def fromInt(i: Int): Option[ResidualMaterial] = i match {
+      case NotSpecified.identity        => Some(NotSpecified)
```

Comment:
It's preferable give `identity` a different name. So there's no ambiguity with `Predef.identity`.